File: app/types/products-search.ts
import { Category } from "@/types/category-model"

export type Product = {
    pid: string
    name: string
    brand: string
    category: Category
    image: string
}

export type ProductsSearchResponse = {
    products: Product[]
    dateUpdated: string
    total: number
    page: number
    pageSize: number
    totalPages: number
    product: string
    category: Category
}

File: services/shared/contracts/products-search.go
package contracts

import "shopping-list/shared/models"

type ProductsSearchResponse struct {
	Products    []models.Product `json:"products"`
	DateUpdated string           `json:"dateUpdated"`
	Total       int              `json:"total"`
	Page        int              `json:"page"`
	PageSize    int              `json:"pageSize"`
	TotalPages  int              `json:"totalPages"`
	Product     string           `json:"product"`
	Category    string           `json:"category"`
}
